### ISSUE TITLE
Add filter Trace Status

### ISSUE THREAD

**Issue by NikolaKalinic** [eclipse-cdt-cloud/theia-trace-extension#1061]
I would like to suggest an enhancement to our trace status functionality by adding a filter option for trace status. Currently, all processes are displayed without any filtering. However, I believe it would be beneficial to introduce a feature that allows users to selectively view specific processes based on their needs.

#### Why a Filter Option Matters:
- **Improved Usability**: With a filter, users can uncheck irrelevant processes and focus on what matters most. This streamlines the trace view and reduces visual clutter.

#### Example:
![example](https://github.com/eclipse-cdt-cloud/theia-trace-extension/assets/93400532/9c307dce-26f5-47b4-838d-e8f68c6e073e)
This is a random example of what it should look like

**Comment by bhufmann** [comment]
Thanks for the interest in Trace Compass. Yes, it would be great to be able to filter (show/hide) rows to be shown. The tree and the timegraph on left are synchronized, that means one line on the tree corresponds to a line on the graph (left). Removing lines have to be done on the tree and on the graph. Enable the checkboxes on the left would allow do deselct rows, but there will needs to be a different control to get the hidden rows back. Maybe having a separate tree dialog (from the menu) where a checkbox tree exists and users can select and deselect threads (rows). 

Interesting would also to be able to select multple rows on the tree, right mouse click and select a hide menu item. This is just a shortcut to hide items.

**Comment by NikolaKalinic** [comment]
First of all, I want to thank you for your response and support for enhancing Trace Compass functionality. Your comment regarding the synchronization of the tree with changes on the graph is extremely important. Allowing users to focus only on essential information, rather than getting confused with unnecessary details, is crucial for improving the user experience.

Thank you once again for your engagement and support.

**Comment by NikolaKalinic** [comment]
@bhufmann I am a student currently working on a student project that utilizes Trace Compass. I would like to add this functionality, but I'm encountering difficulties in following the instructions for starting the project from the README. Could you help me to start the project and provide the necessary support so that I can begin working on this functionality?

**Comment by MatthewKhouzam** [comment]
Hey @NikolaKalinic I want to help out too! :) First off, anything missing in the README?

**Comment by NikolaKalinic** [comment]
Hey @MatthewKhouzam , thank you so much for expressing your willingness to help out! It's fantastic to have support as I dive into this project.

Regarding the README, everything seems clear, but I've encountered some difficulties in getting started. Here's where I stand: I've got Node.js version 18.20, Java 17, and Yarn 1.8 installed on my system. I've also cloned the repository and tried running
 ```bash
yarn tswatch
```
 as instructed, but I seem to be facing some issues.

![e1](https://github.com/eclipse-cdt-cloud/theia-trace-extension/assets/93400532/736bb892-c6df-4493-9621-3dd42992ac9e)

Do you have any insights or suggestions on how to troubleshoot this? Any help would be greatly appreciated!

**Comment by NikolaKalinic** [comment]
Exactly, I only have one repository 'theia-trace-extension'. I thought I could fetch the other necessary components using commands like 'yarn download:server', but it seems that's not how it works the way I imagined. Can you tell me which repositories I need to have and where they should be located? Also, do I need to follow instructions for setting up each one or something else? Can you give me baby steps (rubber ducking) :joy: :joy:  ?

**Comment by MatthewKhouzam** [comment]
Ok, that helps! and Yes, I love the enthusiasm. ;)

Basically, `yarn tswatch` says "Hey, use the local libraries, don't donwload any. 

If you're still at the begining, you have a few choices.
1. clean up your git... (git clean -fdx, git reset hard)
2. Erase everything restart with no yarn tswatch
3. the best, download the other modules like `timeline-chart` and link them.

Hope this helps.

**Comment by NikolaKalinic** [comment]
I've successfully run the project! Thank you so much for your time and the invaluable assistance you've provided to help me through this. Your support mean a lot, and I truly appreciate everything you've done. I'm now ready to move forward with this feature! Once again, a big thank you!

**Comment by MatthewKhouzam** [comment]
Looking forward to seeing the progress!

**Comment by NikolaKalinic** [comment]
Hey @MatthewKhouzam, I need small help again. I run application with yarn start:server and yarn start:browser, but now when I want to change small part of code like add some id on some element, my change is not appeared in DOM, but if I check that in Sources I can see that. Does I need to rebuild something or what?
